.github/workflows/cli.yml

```diff
@@ -73,265 +45,124 @@ jobs:
-        # the action doesn't support it yet (https://github.com/actions-rs/toolchain/issues/208).
-        # Once it does, we should be able to remove the `with` block entirely.
-        with:
-          profile: minimal
-          toolchain: "1.62.1"
-          target: x86_64-pc-windows-msvc
-          override: true
-
-      - name: set up a cache for .cargo
-        uses: actions/cache@v3
-        timeout-minutes: 5
-        continue-on-error: true
-        with:
-          path: ~/.cargo/*
-          key: ${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('Cargo.lock', 'Cargo.toml') }}
-
-      - name: set up a cache for target
-        uses: actions/cache@v3
-        timeout-minutes: 5
-        continue-on-error: true
+      - name: Rust job init
+        uses: ./.github/actions/init_rust_job
         with:
-          path: target
-          key: ${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('Cargo.lock', 'Cargo.toml') }}
+          cdn: ${{ github.ref_name == 'main' && env.PROD_ASSETS_BUCKET || env.DEV_ASSETS_BUCKET }}
+          assets_version: ${{ env.ASSETS_VERSION }}
 
-      - name: fetch assets
-        run: |
-          curl https://temp-artifact-storage.s3.eu-west-3.amazonaws.com/assets.zip --output assets.zip
-          unzip assets.zip
-          rm -r crates/server/assets
-          mv assets crates/server/assets
-          cp -r crates/server/assets ~/.grafbase/
-          touch ~/.grafbase/version.txt
-        shell: bash
-
-      - name: test
+      - name: Test
         uses: actions-rs/cargo@v1
         with:
           command: test
 
-      - name: build
+      - name: Build release
+        if: startsWith(github.ref, 'refs/tags/')
         uses: actions-rs/cargo@v1
-        if: github.ref == 'refs/heads/main'
         with:
           command: build
           args: "--release --target x86_64-pc-windows-msvc"
 
-      - name: archive production artifacts
+      - name: Upload binaries
+        if: startsWith(github.ref, 'refs/tags/')
         uses: actions/upload-artifact@v3
-        if: github.ref == 'refs/heads/main'
         with:
-          name: grafbase-x86_64-pc-windows-msvc.exe
-          path: target/x86_64-pc-windows-msvc/release/grafbase.exe
+          name: ${{ github.ref_name }}-windows
+          path: |
+            target/x86_64-pc-windows-msvc/release/grafbase.exe
 
   linux:
     runs-on: ubuntu-latest
-
+    needs: [lint]
     steps:
-      - name: clone repository
+      - name: Get sources
         uses: actions/checkout@v3
 
-      - name: move CLI to root
-        run: |
-          shopt -s extglob
-          rm -r !("cli")
-          mv cli/* .
-          rm -r cli
-
-      # FIXME: Drop when https://github.com/actions-rs/toolchain/issues/126 is fixed.
-      - name: create `rust-toolchain`
-        run: |
-          grep -Po '(?<=^channel = ")\d+\.\d+(\.\d+)?(?=")' rust-toolchain.toml > rust-toolchain && rm rust-toolchain.toml
```

Comment:
Is this still being handled? I can't find it in the new version. IIRC it didn't work on Windows

Reply:
its building correctly in windows. Not sure what you mean, I'm not using the rustup github action.

Reply:
Oh, so it uses the toolchain automatically then, ok